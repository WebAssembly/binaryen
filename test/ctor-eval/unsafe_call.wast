(module
  (memory 256 256)
  (data (i32.const 10) "waka waka waka waka waka")
  (export "test1" $test1)
  (export "test2" $test2)
  (export "test3" $test3)
  (func $test1
    (call $unsafe-to-call) ;; unsafe to call
    (i32.store (i32.const 12) (i32.const 115)) ;; a safe store, should alter memory
    (i32.store16 (i32.const 20) (i32.const 31353))
    (i32.store8 (i32.const 23) (i32.const 120))
  )
  (func $unsafe-to-call
    (unreachable)
  )
)
