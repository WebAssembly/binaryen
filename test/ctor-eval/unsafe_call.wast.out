(module
 (type $none_=>_none (func))
 (memory $0 256 256)
 (data (i32.const 10) "waka waka waka waka waka")
 (export "test1" (func $test1))
 (func $test1 (; 0 ;)
  (call $unsafe-to-call)
  (i32.store
   (i32.const 12)
   (i32.const 115)
  )
  (i32.store16
   (i32.const 20)
   (i32.const 31353)
  )
  (i32.store8
   (i32.const 23)
   (i32.const 120)
  )
 )
 (func $unsafe-to-call (; 1 ;)
  (unreachable)
 )
)
